src/context/WalletProvider/WalletViewsSwitch.tsx

```diff
@@ -35,20 +34,22 @@ export const WalletViewsSwitch = (props: WalletViewProps) => {
   }
 
   useEffect(() => {
-    if (props.routePath) {
-      history.push(props.routePath as string)
+    if (state?.initalRoute) {
+      history.push(state.initalRoute)
     }
-  }, [history, props.routePath])
+  }, [history, state?.initalRoute])
 
   return (
     <>
       <NativePasswordRequired
         onConnect={(wallet: NativeHDWallet) => {
           dispatch({ type: WalletActions.SET_WALLET, payload: wallet })
           dispatch({ type: WalletActions.SET_IS_CONNECTED, payload: true })
```

Comment:
Can a wallet be set in state and not connected?

Reply:
yeah it can still be on the keyring but not unlocked. but the way we're using it now there wouldnt be a wallet in state if it wasnt connected. so thinking ahead some wallets may require separate dispatches

Reply:
Ok cool just making sure. When I was writing tests I was wondering if that was expected behavior.